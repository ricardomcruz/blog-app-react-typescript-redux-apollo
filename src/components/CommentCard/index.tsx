import classNamesBind from 'classnames/bind';
import React from 'react';
import Card from 'react-bootstrap/Card';
import { RawComment } from '../../_modules/comments/models/Comment';
import styles from './styles.module.scss';

const cx = classNamesBind.bind(styles);

interface CommentCardProps {
  comment: RawComment
}

const CommentCard = (
  {
    comment: {
      date,
      content,
      user
    }
  }: CommentCardProps
) => {

  return (
      <Card className={cx("comment-card", "my-3")}>
      <Card.Header className={cx("comment-card-header")}>
        <b>{user}</b>
        <div>
          <p>{date}</p>
        </div>
      </Card.Header>
        <Card.Body >
          <Card.Text>
            {content}
          </Card.Text>
        </Card.Body>
      </Card >
  );
}

export default CommentCard;
