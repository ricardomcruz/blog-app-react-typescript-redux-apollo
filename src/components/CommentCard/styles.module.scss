// Required
@import "../../globals.scss";

.comment-card-header {
  display: flex;
  justify-content: space-between;
}