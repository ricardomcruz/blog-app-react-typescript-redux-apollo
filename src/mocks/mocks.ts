
import { Comment, CommentDTO } from '../../_modules/comments/models/Comment';
import { Post } from '../../_modules/posts/models/Post';

export const postMock: Post = {
  id: 1,
  title: 'title',
  author: 'author',
  publish_date: 'publish_date',
  description: 'description',
  content: 'content'
}

export const commentMockDTO: CommentDTO = {
  postId: 2,
  user: 'user',
  date: 'date',
  content: 'content',
};

export const commentMock: Comment = {
  ...commentMockDTO,
  id: 1,
  parent_id: 2,
};

export const postWithCommentsMock: Post = {
 ...postMock,
 comments: commentMock
};