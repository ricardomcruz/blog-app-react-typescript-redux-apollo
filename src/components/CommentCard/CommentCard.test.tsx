/** * @jest-environment jsdom */

import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { commentMock } from '../../mocks/mocks';
import CommentCard from './index';

describe('CommentCard', () => {

  it('should render the Article Card', () => {
    const { asFragment }= render(
      <CommentCard
        comment={commentMock}
        darkMode
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
