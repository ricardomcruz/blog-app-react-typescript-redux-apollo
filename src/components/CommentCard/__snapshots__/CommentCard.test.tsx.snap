// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard should render the Article Card 1`] = `
<DocumentFragment>
  <div
    class="comment-card my-3 bg-dark card"
  >
    <div
      class="comment-card-header text-light bg-secondary card-header"
    >
      <small>
        <b>
          user
        </b>
      </small>
      <div>
        <small>
          date
        </small>
      </div>
    </div>
    <div
      class="card-body"
    >
      <p
        class="card-text"
      >
        content
      </p>
    </div>
  </div>
</DocumentFragment>
`;
